File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        int max = 0;
        
        while(true) {
        	StringTokenizer st = new StringTokenizer(br.readLine());
        	
        	int fir = Integer.parseInt(st.nextToken());
        	int sec = Integer.parseInt(st.nextToken());
        	int thr = Integer.parseInt(st.nextToken());
        	
        	if(fir == 0 && sec == 0 && thr == 0)
				break;
        	
        	max = Math.max(Math.max(fir, sec),thr);
        	int sum = 0;
        	
        	if(max == fir) {
        		sum = sec + thr;
        	} else if(max == sec) {
        		sum = fir + thr;
        	} else {
        		sum = fir + sec;
        	}
        	
        	if(max >= sum) {
        		System.out.println("Invalid");
        	} else {
        		if(fir == sec && sec == thr) {
        			System.out.println("Equilateral");
        		} else {
        			if(fir == sec || fir == thr || sec == thr) {
        				System.out.println("Isosceles");
        			} else {
        				System.out.println("Scalene");
        			}
        		}
        	}
        	
        
        }
        
        
    }
}

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.cc
#include <iostream>
#include <cmath>

using namespace std;

int main()
{	
	int maxA = 0;
	int angle1, angle2, angle3;

	while (true) {
		cin >> angle1 >> angle2 >> angle3;

		if (angle1 == 0 && angle2 == 0 && angle3 == 0)
			break;

		maxA = max(max(angle1, angle2), angle3);
		int sum = 0;

		if (maxA == angle1) {
			sum = angle2 + angle3;
		}
		else if (maxA == angle2) {
			sum = angle1 + angle3;
		}
		else {
			sum = angle1 + angle2;
		}

		if (maxA >= sum) {
			cout << "Invalid" << "\n";
		}
		else {
			if (angle1 == angle2 && angle2 == angle3) {
				cout << "Equilateral" << "\n";
			}
			else {
				if (angle1 == angle2 || angle1 == angle3 || angle2 == angle3) {
					cout << "Isosceles" << "\n";
				}
				else {
					cout << "Scalene" << "\n";
				}
			}
		}

	}
	
	
	return 0;
}